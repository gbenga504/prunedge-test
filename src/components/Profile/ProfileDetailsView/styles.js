import { fontWeight, fonts, fontsize, commonCss, color } from "../../../Css";

export const styles = {
  container: {
    ...commonCss.flexColumn,
    padding: 22,
    background: "#fff",
    width: "46%",
    height: "max-content",
    boxShadow: "0px 2px 24px rgba(0, 0, 0, 0.04)"
  },
  title: {
    ...commonCss.title
  },
  marker: {
    ...commonCss.titleMarker
  },
  personalDetailsContainer: {
    ...commonCss.flexColumn,
    alignItems: "center",
    marginTop: 41
  },
  avatarContainer: {
    ...commonCss.profileAvatar
  },
  userName: {
    ...commonCss.profileUserName
  },
  role: {
    ...commonCss.profileUserRole
  },
  separator: {
    height: 1,
    marginTop: 11,
    marginBottom: 23,
    background: color.separator,
    width: "100%"
  },
  organizationContainer: {
    ...commonCss.flex,
    justifyContent: "space-between",
    "& .profile__organization-single-unit": {
      ...commonCss.flexColumn,
      alignItems: "center"
    }
  },
  iconContainer: {
    ...commonCss.flex,
    justifyContent: "center",
    borderRadius: "50%",
    height: 35,
    width: 35,
    border: `3px solid ${color.lightBlueGrey}`,
    background: color.blueGrey,
    "& svg": {
      color: "#fff",
      fontSize: 15
    }
  },
  lightGreyText: {
    marginTop: 5,
    fontWeight: fontWeight.regular,
    color: color.lightGrey,
    fontFamily: fonts.main,
    fontSize: fontsize.small
  },
  darkText: {
    marginTop: 12,
    fontWeight: fontWeight.bold,
    color: color.dark,
    fontFamily: fonts.main,
    fontSize: fontsize.base
  },
  uploadContainer: {
    ...commonCss.flex,
    background: "#fff",
    position: "absolute",
    width: 32,
    height: 32,
    marginTop: -20,
    borderRadius: "50%",
    marginLeft: 60,
    justifyContent: "center",
    boxShadow: `0px 0px 15px 0px ${color.lightBlueGrey}`,
    "& svg": {
      fontSize: 15,
      color: color.dark
    },
    "& input": {
      width: 20,
      height: 20,
      position: "absolute",
      opacity: 0
    }
  }
};
