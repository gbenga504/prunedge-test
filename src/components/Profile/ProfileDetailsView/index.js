import React from "react";
import withStyles from "@material-ui/core/styles/withStyles";
import Typography from "@material-ui/core/Typography";
import AccountingIcon from "@material-ui/icons/DeviceHub";
import GroupIcon from "@material-ui/icons/Group";
import EditIcon from "@material-ui/icons/Edit";

import { styles } from "./styles";
import { color } from "../../../Css";

class ProfileDetailsVew extends React.PureComponent {
  state = {
    avatar:
      "https://media.licdn.com/dms/image/C5603AQHnNE6t21CVkg/profile-displayphoto-shrink_100_100/0?e=1571875200&v=beta&t=DOFMyno8Ei2tZmecWIppdRxDcicR43lX-0k2Q7rktaU"
  };

  handleFileChange = e => {
    let file = e.target.files[0],
      fileReader = new FileReader();

    if (file) {
      fileReader.onload = ev => {
        this.setState({
          avatar: ev.target.result
        });
      };
      fileReader.readAsDataURL(file);
    }
  };

  renderPersonalDetails = () => {
    let { classes } = this.props;

    return (
      <div className={classes.personalDetailsContainer}>
        <div className={classes.avatarContainer}>
          <img alt="user" src={this.state.avatar} />
          <div className={classes.uploadContainer}>
            <EditIcon />
            <input type="file" onChange={this.handleFileChange} />
          </div>
        </div>
        <Typography variant="body2" classes={{ body2: classes.userName }}>
          Joahn Samuel
        </Typography>
        <Typography variant="body2" classes={{ body2: classes.role }}>
          Front Desk Officer
        </Typography>
      </div>
    );
  };

  renderOrganizationDetails = () => {
    let { classes } = this.props;

    return (
      <div className={classes.organizationContainer}>
        <div className="profile__organization-single-unit">
          <div
            className={classes.iconContainer}
            style={{ borderColor: color.lightPurple, background: color.purple }}
          >
            <AccountingIcon />
          </div>
          <Typography
            variant="body2"
            classes={{ body2: classes.lightGreyText }}
          >
            Department
          </Typography>
          <Typography variant="body2" classes={{ body2: classes.darkText }}>
            Accounting
          </Typography>
        </div>
        <div className="profile__organization-single-unit">
          <div className={classes.iconContainer}>
            <GroupIcon />
          </div>
          <Typography
            variant="body2"
            classes={{ body2: classes.lightGreyText }}
          >
            Unit
          </Typography>
          <Typography variant="body2" classes={{ body2: classes.darkText }}>
            Finance
          </Typography>
        </div>
        <div className="profile__organization-single-unit">
          <div className={classes.iconContainer}>
            <GroupIcon />
          </div>
          <Typography
            variant="body2"
            classes={{ body2: classes.lightGreyText }}
          >
            Organization
          </Typography>
          <Typography variant="body2" classes={{ body2: classes.darkText }}>
            Prunedge
          </Typography>
        </div>
      </div>
    );
  };

  render() {
    let { classes } = this.props;

    return (
      <div className={classes.container}>
        <div style={{ position: "relative", width: "auto" }}>
          <Typography variant="body1" classes={{ body1: classes.title }}>
            Profile
          </Typography>
          <span className={classes.marker} />
        </div>
        {this.renderPersonalDetails()}
        <div className={classes.separator} />
        {this.renderOrganizationDetails()}
      </div>
    );
  }
}

export default withStyles(styles)(ProfileDetailsVew);
