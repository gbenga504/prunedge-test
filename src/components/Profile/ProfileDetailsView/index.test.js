import React from "react";
import { create } from "react-test-renderer";

import ProfileDetailsView from "./index";

describe("ProfileDetailsView", () => {
  it("renders the profile details view", () => {
    const tree = create(<ProfileDetailsView />);
    expect(tree).toMatchSnapshot();
  });
});
