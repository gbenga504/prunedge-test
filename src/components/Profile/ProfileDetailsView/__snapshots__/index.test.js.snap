// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileDetailsView renders the profile details view 1`] = `
<div
  className="ProfileDetailsVew-container-1"
>
  <div
    style={
      Object {
        "position": "relative",
        "width": "auto",
      }
    }
  >
    <p
      className="MuiTypography-root MuiTypography-body1 ProfileDetailsVew-title-2"
    >
      Profile
    </p>
    <span
      className="ProfileDetailsVew-marker-3"
    />
  </div>
  <div
    className="ProfileDetailsVew-personalDetailsContainer-4"
  >
    <div
      className="ProfileDetailsVew-avatarContainer-5"
    >
      <img
        alt="user"
        src="https://media.licdn.com/dms/image/C5603AQHnNE6t21CVkg/profile-displayphoto-shrink_100_100/0?e=1571875200&v=beta&t=DOFMyno8Ei2tZmecWIppdRxDcicR43lX-0k2Q7rktaU"
      />
      <div
        className="ProfileDetailsVew-uploadContainer-13"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
          />
        </svg>
        <input
          onChange={[Function]}
          type="file"
        />
      </div>
    </div>
    <p
      className="MuiTypography-root MuiTypography-body2 ProfileDetailsVew-userName-6"
    >
      Joahn Samuel
    </p>
    <p
      className="MuiTypography-root MuiTypography-body2 ProfileDetailsVew-role-7"
    >
      Front Desk Officer
    </p>
  </div>
  <div
    className="ProfileDetailsVew-separator-8"
  />
  <div
    className="ProfileDetailsVew-organizationContainer-9"
  >
    <div
      className="profile__organization-single-unit"
    >
      <div
        className="ProfileDetailsVew-iconContainer-10"
        style={
          Object {
            "background": "#B073FD",
            "borderColor": "#D0ABFE",
          }
        }
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M17 16l-4-4V8.82C14.16 8.4 15 7.3 15 6c0-1.66-1.34-3-3-3S9 4.34 9 6c0 1.3.84 2.4 2 2.82V12l-4 4H3v5h5v-3.05l4-4.2 4 4.2V21h5v-5h-4z"
          />
        </svg>
      </div>
      <p
        className="MuiTypography-root MuiTypography-body2 ProfileDetailsVew-lightGreyText-11"
      >
        Department
      </p>
      <p
        className="MuiTypography-root MuiTypography-body2 ProfileDetailsVew-darkText-12"
      >
        Accounting
      </p>
    </div>
    <div
      className="profile__organization-single-unit"
    >
      <div
        className="ProfileDetailsVew-iconContainer-10"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M16 11c1.66 0 2.99-1.34 2.99-3S17.66 5 16 5c-1.66 0-3 1.34-3 3s1.34 3 3 3zm-8 0c1.66 0 2.99-1.34 2.99-3S9.66 5 8 5C6.34 5 5 6.34 5 8s1.34 3 3 3zm0 2c-2.33 0-7 1.17-7 3.5V19h14v-2.5c0-2.33-4.67-3.5-7-3.5zm8 0c-.29 0-.62.02-.97.05 1.16.84 1.97 1.97 1.97 3.45V19h6v-2.5c0-2.33-4.67-3.5-7-3.5z"
          />
        </svg>
      </div>
      <p
        className="MuiTypography-root MuiTypography-body2 ProfileDetailsVew-lightGreyText-11"
      >
        Unit
      </p>
      <p
        className="MuiTypography-root MuiTypography-body2 ProfileDetailsVew-darkText-12"
      >
        Finance
      </p>
    </div>
    <div
      className="profile__organization-single-unit"
    >
      <div
        className="ProfileDetailsVew-iconContainer-10"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M16 11c1.66 0 2.99-1.34 2.99-3S17.66 5 16 5c-1.66 0-3 1.34-3 3s1.34 3 3 3zm-8 0c1.66 0 2.99-1.34 2.99-3S9.66 5 8 5C6.34 5 5 6.34 5 8s1.34 3 3 3zm0 2c-2.33 0-7 1.17-7 3.5V19h14v-2.5c0-2.33-4.67-3.5-7-3.5zm8 0c-.29 0-.62.02-.97.05 1.16.84 1.97 1.97 1.97 3.45V19h6v-2.5c0-2.33-4.67-3.5-7-3.5z"
          />
        </svg>
      </div>
      <p
        className="MuiTypography-root MuiTypography-body2 ProfileDetailsVew-lightGreyText-11"
      >
        Organization
      </p>
      <p
        className="MuiTypography-root MuiTypography-body2 ProfileDetailsVew-darkText-12"
      >
        Prunedge
      </p>
    </div>
  </div>
</div>
`;
